Gems/ROS2/Code/Include/ROS2/Manipulator/MotorizedJointComponent.h

```diff
@@ -0,0 +1,97 @@
+    //! @note This class is already used through ROS2FrameComponent.
+    // TODO This is a prototype. Tasks: refactor, add bus interface, rotation, ramps, cascading controllers, tests.
+    class MotorizedJointComponent
+        : public AZ::Component
+        , public AZ::TickBus::Handler
+        , public MotorizedJointRequestBus::Handler
+    {
+    public:
+        AZ_COMPONENT(MotorizedJointComponent, "{AE9207DB-5B7E-4F70-A7DD-C4EAD8DD9403}", AZ::Component);
+
+        MotorizedJointComponent() = default;
+        ~MotorizedJointComponent() = default;
+        void Activate() override;
+        void Deactivate() override;
+        static void Reflect(AZ::ReflectContext* context);
+
+        //! Set a setpoint (e.g. desired local position). The controller will follow it.
+        void SetSetpoint(float setpoint) override;
+
+        //! Get a setpoint
+        float GetSetpoint() override;
+
+        //! Get current control error. It is the difference between control value and measurement.
+        //! When the setpoint is reached this should be close to zero.
+        //! @returns control error, in meters for linear joints and in radians for angular joints.
+        float GetError() override;
+
+        //! Get current position from measurement.
+        //! @returns current position, in meters for linear joints and radians for angular joints.
+        float GetCurrentMeasurement() override;
+
+        //! Get a degree of freedom direction.
+        //! @returns direction of joint movement in global coordinates.
+        AZ::Vector3 GetDir() const
+        {
+            return m_jointDir;
+        };
+
+    private:
+        float ComputeMeasurement(AZ::ScriptTimePoint time);
+        void SetVelocity(float velocity, float deltaTime);
+        void ApplyLinVelAnimation(float velocity, float deltaTime);
+        void ApplyLinVelRigidBodyImpulse(float velocity, float deltaTime);
+        void ApplyLinVelRigidBody(float velocity, float deltaTime);
+        void OnTick(float deltaTime, AZ::ScriptTimePoint time) override;
+
+        AZ::Vector3 m_jointDir{ 0.f, 0.f, 1.f }; //!< Direction of joint movement in parent frame of reference, used to compute measurement.
+        AZ::Vector3 m_effortAxis{ 0.f, 0.f, 1.f }; //!< Direction of force or torque application in owning entity frame of reference.
+        AZStd::pair<float, float> m_limits{ -0.5f, 0.5f }; //!< limits of joint, the force is applied only when joint is within limits.
+        VehicleDynamics::PidConfiguration m_pidPos; //!< PID controller for position.
+
+        bool m_linear{ true }; //!< Linear mode. The force is applied through RigidBodyBus.
+        bool m_animationMode{ true }; //!< Use TransformBus (animation mode, no physics) instead of RigidBodyBus.
+
+        // TODO - remove test signal?
+        bool m_testSinusoidal{ true }; //!< Enable sinusoidal signal generator to setpoint (for tuning).
```

Comment:
Should this TODO be done now: https://github.com/o3de/sig-core/blob/main/governance/Coding-Standards-and-Style-Guide.md#todo-comments

Reply:
Removed, raised https://github.com/RobotecAI/o3de-ros2-gem/issues/339